llvm/tools/sycl-post-link/SYCLDeviceRequirements.cpp

```diff
@@ -69,6 +67,27 @@ void llvm::getSYCLDeviceRequirements(
         std::vector<uint32_t>(Values.begin(), Values.end());
   }
 
+  std::optional<llvm::SmallVector<uint64_t, 3>> ReqdWorkGroupSize;
+  for (const Function &F : MD.getModule()) {
+    if (const MDNode *MDN = F.getMetadata("reqd_work_group_size")) {
+      llvm::SmallVector<size_t, 3> NewReqdWorkGroupSize;
+      for (size_t I = 0, E = MDN->getNumOperands(); I < E; ++I)
+        NewReqdWorkGroupSize.push_back(
+            ExtractUnsignedIntegerFromMDNodeOperand(MDN, I));
+      if (!ReqdWorkGroupSize)
```

Comment:
It looks like now we save only the first occurrence of the `reqd_work_group_size` metadata, isn't it?